### PR TITLE
COD-2/COD-4

### PR DESCRIPTION
**Pull request by zohaib194**

### CODE REVIEW

File: backend/apiServer/model/mongodb.go [hunk 1]
```diff
@@ -101,3 +101,29 @@ func (db *MongoDB) findRepoByURI(uri string) (repo RepoModel, err error) {
...
+
+	if !bson.IsObjectIdHex(id) {
+		err = errors.New("Invalid id")
+		log.Println(logError+"Id received is incorrect: ", err)
+		return RepoModel{}, err
+	}
+
+	keyID := bson.ObjectIdHex(id)
+
+	// Retrun empty repo with error if error is not "Not found"
```
Comment (Xillez) [inline backend/apiServer/model/mongodb.go]: Typo "retrun" -> "return"

File: backend/apiServer/controller/repo.go [hunk 2]
```diff
@@ -88,3 +90,87 @@ func (repo RepoController) NewRepoFromURI(w http.ResponseWriter, r *http.Request
...
+/**
+* @api {Post} /repo/:id Parse the repository assosiated with id.
+* @apiName Parse repository.
+* @apiGroup Repository
+* @apiPermission none
+*
+* @apiParam {String} Id Id of submitted git repository.
+*
+* @apiParamExample {url} Parse repository:
+* 	/repo/:id
```
Comment (eldarht) [inline backend/apiServer/controller/repo.go]: @apiParamExample should be an example input (http://apidocjs.com/#param-api-param-example).
/repo/:id is given by @apiParam.

File: backend/apiServer/controller/repo.go [hunk 3]
```diff
@@ -88,3 +90,87 @@ func (repo RepoController) NewRepoFromURI(w http.ResponseWriter, r *http.Request
...
+		exstRepo, err := model.RepoModel{}.GetRepoByID(id)
+
+		if err != nil {
+			http.Error(w, "Invalid parameters", http.StatusBadRequest)
+			log.Println("Could not find repo in db: ", err.Error())
+			return
+		}
+
+		// List all files in the repository directory.
+		files, err := exstRepo.GetRepoFile(model.RepoPath + "/" + exstRepo.ID.Hex())
```
Comment (eldarht) [inline backend/apiServer/controller/repo.go]: exstRepo contains id.Hex() given to it as a parameter and model.Repopath is here taken from model into controller and sendt back into model. could just have getrepo use models local variables and exstRepo that it works on (no param needed)

File: backend/apiServer/controller/repo.go [hunk 4]
```diff
@@ -88,3 +90,87 @@ func (repo RepoController) NewRepoFromURI(w http.ResponseWriter, r *http.Request
...
+
+		if err != nil {
+			http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+			log.Println("Could not find files error: ", err.Error())
+			return
+		}
+
+		// Fetch all fuctions given in files.
+		functions, err := exstRepo.ParseFunctionsFromFiles(files, exstRepo)
```
Comment (eldarht) [inline backend/apiServer/controller/repo.go]: exstRepo is sendt into exstRepo. This functionality could be usefull but as an overloaded function. default without parameters would probably be better.

**Pull request merged**